Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/ApiPropertyMappingEnhancer.ts

```diff
@@ -37,18 +39,21 @@ function parentPrefixRemovalConvention(property: EntityProperty): string {
  * API descriptor reference property names are suffixed with "Descriptor"
  */
 function apiDescriptorReferenceName(property): string {
-  return normalizeDescriptorSuffix(uncapitalize(property.fullPropertyName));
+  return normalizeDescriptorSuffix(uncapitalize(property.metaEdName));
 }
 
 /**
  * The basic name of a property in the API.  Generally, the property full name with the
  * first character lower cased, and pluralized if an array.
  */
-function apiFullName(property: EntityProperty, { removeCollectionPrefixes }): string {
-  if (property.isCollection && removeCollectionPrefixes) {
+function apiFullName(property: EntityProperty, { removePrefixes }: NamingOptions): string {
+  if (property.isCollection && removePrefixes) {
     return uncapitalize(pluralize(parentPrefixRemovalConvention(property)));
   }
-  if (property.isCollection && !removeCollectionPrefixes) {
+  if (property.type === 'common' && removePrefixes) {
+    return uncapitalize(parentPrefixRemovalConvention(property));
```

Comment:
Needed to remove prefixes on all commons e.g. Assessment.AssessmentPeriod -> "period" not "assessmentPeriod"